table.go

```diff
@@ -1339,63 +1339,121 @@ func (t *TableBlock) abortCompaction(granule *Granule) {
 	}
 }
 
-func (t *TableBlock) Serialize(writer io.Writer) error {
-	ctx := context.Background()
+func mergeDynCols(a, b map[string][]string) map[string][]string {
+	for key, vals := range b {
+		a[key] = append(a[key], vals...)
+	}
 
-	// Read all row groups
-	rowGroups := []dynparquet.DynamicRowGroup{}
+	return bufutils.Dedupe(a)
+}
 
-	rowGroupsChan := make(chan any)
+/*
+	Serialize the table block into a single Parquet file.
+	The Serialize function will walk all Granules in the block, compact each Granule into a sorted set of Row groups, then write those
+	row groups to the final Parquet file, repeating for each Granule. This leverages the fact that the index has alreayd sorted the Granules
+	so no need to sort them further than the intra-granule parts.
+*/
+func (t *TableBlock) Serialize(writer io.Writer) error {
+	dynCols := map[string][]string{} // NOTE: it would be nice if we didn't have to go back and find all the dynamic columns...
```

Comment:
Agreed! At some point, these dynamic columns should be a stateful part of the schema. Would make execution code simpler too. Created an issue: https://github.com/polarsignals/frostdb/issues/366

Reply:
Awesome thank you!